code/measures/cosine_similarity.py

```diff
@@ -1,45 +1,39 @@
 import nml_parseutil
 from article import Article
 from measure_constants import MeasureConstants
+from bow_similarity import bow_similarity_score
 
 import numpy as np
 import datetime
 from nltk.stem import PorterStemmer 
 from nltk.tokenize import word_tokenize
 from nltk.corpus import stopwords
-from nltk import ngrams, FreqDist
+
+from sklearn.feature_extraction.text import TfidfVectorizer
 
 stop_words = set(stopwords.words('english'))
 ps = PorterStemmer()
+vec = TfidfVectorizer()
 
 class CosineSimilarity:
 
-    def __init__(self, measure_const = MeasureConstants()):
+    def __init__(self, measure_const = MeasureConstants(STORAGE_MODE = "string")):
         self.measure_const = measure_const
 
-    def term_frequency(article_text):
-    	"""
-    	Arguments:
-    		article_text: A set containing all of the stemmed words in an article.
-
-    	Returns: a vector representing term frequencies for each word in the article.
-
-
-    	"""
 
     def stem_and_filter(article):
```

Comment:
If these functions are all members of the class, it would seem they would like to have access to __self__... It also looks like measure_const would not be available in places if you don't pass self, which is an issue I was running into with your code.